tests/fetch_body_test.go

```diff
@@ -706,6 +706,76 @@ func TestFetchBodyPeekInvalidSectionNumber(t *testing.T) {
 	})
 }
 
+func TestFetchBody_Dovecot_InvalidMessageHeader(t *testing.T) {
+	// This tests fails when requesting when fetch BODY.PEEK[HEADER.FIELDS (In-Reply-To In-Reply-To Cc)].
+	// Instead of only returning In-Reply-To and Cc, it was also returning the References header.
+	const message = `From tss@iki.fi  Mon Apr  7 01:27:38 2003
+Received: with ECARTIS (v1.0.0; list dovecot); Mon, 07 Apr 2003 01:27:38 +0300 (EEST)
+Return-Path: <tss@iki.fi>
+X-Original-To: dovecot@procontrol.fi
+Delivered-To: dovecot@procontrol.fi
+Received: from oma.irssi.org (ip213-185-36-189.laajakaista.mtv3.fi [213.185.36.189])
+	by danu.procontrol.fi (Postfix) with ESMTP id 2A7282387F
+	for <dovecot@procontrol.fi>; Mon,  7 Apr 2003 01:27:38 +0300 (EEST)
+Received: by oma.irssi.org (Postfix, from userid 1000)
+	id 3D1565E01F95; Mon,  7 Apr 2003 01:27:36 +0300 (EEST)
+Subject: [dovecot] Re: message order reversed on copying
+From: Timo Sirainen <tss@iki.fi>
```

Comment:
I am not 100% sure but this should be also valid header. 

```
From: "Timo =?UTF-8?B?VGhpcyBpcyBmdW5ueSBzdG9yeSBidXQgaXQgaXMgcG9zc2libGUgdG8gcHV0IHF1aXRlIGEgbG9u
ZyB0ZXh0IGludG8gdGhlIHN1YmplY3QgZmllbGQuLiBzbyBsb25nIHRoYXQgYWN0dWFsbHkgbmVl
ZGVkIHRvIGxpbmUgYnJlYWsgaW5zaWRlIHRoZSBxdW90YXRpb25zLg==?= Sirainen" <tss@iki.fi>
```

The idea is that when quotation is opened you don't need to put `\t` in new lines. I know it is used for long attachment names in MIME header. If it is applicable for `From:` field I am not sure.